Letterbook.Adapter.ActivityPub/Client.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Net.Http.Json;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Text.Json;
+using AutoMapper;
+using Letterbook.ActivityPub;
+using Letterbook.Adapter.ActivityPub.Exceptions;
+using Letterbook.Adapter.ActivityPub.Mappers;
+using Letterbook.Adapter.ActivityPub.Signatures;
+using Letterbook.Core.Adapters;
+using Letterbook.Core.Models;
+using Letterbook.Core.Values;
+using Microsoft.Extensions.Logging;
+using Profile = AutoMapper.Profile;
+
+namespace Letterbook.Adapter.ActivityPub;
+
+public class Client : IActivityPubClient, IActivityPubAuthenticatedClient, IDisposable
+{
+    private readonly ILogger<Client> _logger;
+    private readonly HttpClient _httpClient;
+    private readonly HttpRequestMessage _message;
+    private Models.Profile? _profile = default;
+    
+    private static readonly IMapper ProfileMapper = new Mapper(ProfileMappers.DefaultProfile);
+    private static readonly IMapper AsApMapper = new Mapper(Mappers.AsApMapper.Config);
+
+    public Client(ILogger<Client> logger, HttpClient httpClient)
+    {
+        _logger = logger;
+        _httpClient = httpClient;
+        _message = new HttpRequestMessage();
+    }
+    
+    [SuppressMessage("ReSharper", "ExplicitCallerInfoArgument")]
+    // Stream saves on allocations
+    private static async Task<Stream?> ReadResponse(HttpResponseMessage response, 
+        [CallerMemberName] string name="",
```

Comment:
Yeah, I'm going to write some things about exception handling generally. Those attributes get autopopulated with meta information about the calling site by the compiler (otherwise we would have to use reflection for that). My thinking is that we can build Exceptions with the kind of info that will help with offsite debugging, since at some point letterbook will be run by people who are not us.

Reply:
Wouldn't that be included in the stack trace by default?

Reply:
Yes, but I'm not sure logging full stack traces is a good idea

Reply:
Same, let's discuss as its own topic in #114